test/test_linalg.py

```diff
@@ -4951,6 +4951,30 @@ def gen_matrices():
                         self.assertEqual(B_left, X @ A_adj)
 
 
+    @onlyCPU
+    @dtypes(*floating_and_complex_types())
+    def test_linalg_lu_cpu_errors(self, device, dtype):
+        sample = torch.randn(3, 2, 2, device=device, dtype=dtype)
+        B = torch.randn(3, 2, 2, device=device, dtype=dtype)
```

Comment:
OK, these are square inputs, so the mismatch between row/column pivoting got undetected. Let's add rectangular shapes for completeness as well?